src/mlio/record_readers/detail/text_line.cxx

```diff
@@ -29,7 +32,9 @@ inline namespace v1 {
 namespace detail {
 
 std::optional<record>
-read_line(memory_slice &chunk, bool ignore_leftover)
+read_line(memory_slice &chunk,
+          bool ignore_leftover,
+          std::optional<size_t> max_line_length)
```

Comment:
In a couple places I see that you use `size_t` instead of `std::size_t`. Please make sure that you overall use `std::size_t` to be consistent with the rest of the code.